python/ray/data/read_api.py

```diff
@@ -2668,6 +2668,98 @@ def from_arrow_refs(
+
+            import ray
+
+            ds = ray.data.read_delta_sharing_tables(
+                url=f"{current_address}/delta_share.json#share-with-coinbase-ray-cluster.\
+                        ml_adhoc_data_sharing.ralph_user_pairs_with_seq_features_310d_p01_train",
+                limit=100000,
+                version=1,
+            ).materialize()
+
+    Args:
+        url: A URL under the format "<profile>#<share>.<schema>.<table>".
+        limit: (Optional) A non-negative integer. Load only the ``limit`` rows if the
+            parameter is specified. Use this optional parameter to explore the shared table without
+            loading the entire table into memory.
+        version: (Optional) A non-negative integer. Load the snapshot of the table at
+            the specified version.
+        timestamp: (Optional) A timestamp to specify the version of the table to read.
+        jsonPredicateHints: (Optional) Predicate hints to be applied to the table. For more details
+            , see:
+            https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md#json-predicates-for-filtering.
+        parallelism: This argument is deprecated. Use ``override_num_blocks`` argument.
+        ray_remote_args: kwargs passed to :meth:`~ray.remote` in the read tasks.
+        concurrency: The maximum number of Ray tasks to run concurrently. Set this
+            to control the number of tasks to run concurrently. This doesn't change the
+            total number of tasks run or the total number of output blocks. By default,
+            concurrency is dynamically decided based on the available resources.
+        override_num_blocks: Override the number of output blocks from all read tasks.
+            By default, the number of output blocks is dynamically decided based on
+            input data size and available resources. You shouldn't manually set this
+            value in most cases.
+
+    Returns:
+        A :class:`Dataset` containing the queried data.
+
+    Raises:
+        ValueError: If the URL is not properly formatted or if there is an issue
+            with the Delta Sharing table connection.
+    """
+    from .delta_sharing_datasource import (
+        DeltaSharingDatasource,
+        setup_delta_sharing_connections,
+    )
+
+    table, rest_client = setup_delta_sharing_connections(url)
+
+    datasource = DeltaSharingDatasource(
+        table=table,
+        rest_client=rest_client,
+        jsonPredicateHints=jsonPredicateHints,
+        limit=limit,
+        version=version,
+        timestamp=timestamp,
+    )
+    # DeltaSharing limit is at the add_files level, it will not return
+    # exactly the limit number of rows but it will return less files and rows.
+    return ray.data.read_datasource(
+        datasource=datasource,
+        parallelism=parallelism,
+        ray_remote_args=ray_remote_args,
+        concurrency=concurrency,
+        override_num_blocks=override_num_blocks,
+    ).limit(limit)
```

Comment:
let's not call `limit` here. It may cause suprising issue for other users.

Reply:
I see.
The reason I put a limit here is because as mentioned before, setting a limit number info upfront wont make sure the generated data will have the exact number described in the limit. Is that okay?